text/0000-update_asset_packaging.md

```diff
@@ -0,0 +1,102 @@
+- Feature Name: update_asset_packaging
+- Start Date: 2016-05-10
+- RFC PR:
+- Issue:
+
+# Summary
+[summary]: #summary
+
+Several suggestions have been made to improve the way we handle our assets packaging.
+This RFC attempts to present the various options so that a decision can be made.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, most assets (js/css) are used as gems and managed by bundler.
+A few are included directly in the vendor/ directory due to lack of gems.
+Some of the reasons that we may want to change the way we currently handle this:
+* Some assets are not packaged as gems, so we need to either vendor or package them ourselves.
+* Some asset gems are not updated at the same rate as the upstream source.
+* Each gem dependency must be packaged separately as RPM/DEB.
+* Difficult to integrate newer JS technology such as es2015 or js modules.
+* Assets included in the vendor/ directory need manual updating when new versions are released.
+
+
+# Detailed design
+[design]: #detailed-design
+
+This will depend on the alternative that will be chosen and updated later.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Some reasons to stay with the current solution:
+* No extra work needed.
+* Single package management system - bundler. All dependencies are in one central location, build process is simple.
+* Rails asset pipeline doesn’t require any modifications to handle the current solution.
+
+# Alternatives
+[alternatives]: #alternatives
+
+__Option 1: Stay with current solution__
+
+See above for reasons for and against.
+
+__Option 2: Webpack + npm__
+
+[POC PR](https://github.com/theforeman/foreman/pull/3433)
+
+_Pros:_
+* Live reload development server.
+* Ability to package the code in separate chunks that are only loaded when needed.
+* Multiple loaders allow for server-side tasks such as transpiling, linting, minifing etc. to be run without requiring an additional task runner.
+* Can possibly replace rails asset pipeline in the future if desired.
```

Comment:
Is it? I'm not sure that all the other options allow for a complete replacement including cache-busting, minification, etc.